### DISCUSSION TITLE
How should mendeleev deal with units of properties?

### DISCUSSION THREAD

**Discussion by lmmentel** [lmmentel/mendeleev#162] [Ideas]
Many properties available in `mendeleev` next to numerical values have also units of measurement. The units for most properties are documented in the [data page](https://mendeleev.readthedocs.io/en/stable/data.html) in the docs. All units except for isotope half life are consistent across all value for a given property. This means for example that all boiling points are expressed in Kelvins and covalent radii are in picometers. 

Thanks to recent changes #156 units are now programmatically available  through `mendeleev.models.PropertyMetadata` and therefore can be used to introduce an interface where quantities are returned with explicit unit.

One popular option for modelling units explicitly would be by incorporating the [pint](https://github.com/hgrecco/pint) package for handling untis.

## Open design questions:

- should mendeleev adopt explicit unit treatment for dimensional properties?
- should there be 2 interfaces available, with and without explicit units?
- how would a good interface/api with explicit unit treatment look like? 

## Context

- previous discussion https://github.com/lmmentel/mendeleev/issues/94

**Comment by kalvdans** [comment]
Thanks for bringing up the discussion! I think `mendeleev` should always return SI units, but might provide a library to print a value in SI units to a human-readable string.

**Comment by lmmentel** [comment]
First draft of the units handling is implemented in #239. Right now only `stored` properties are handled and none of the computed properties are covered. The units are defined in `PropertyMetadata` model and then loaded together with values to `pint.Quantity` object when the regular attributes end with `_u` suffix. The logic is handled by `UnitMixin` and should be fully backwards compatible and return pure `float` or `int` for attributes without the `_u` suffix.

## Example

```python
from mendeleev import H
# without units
H.atomic_radius
25.0

# with units
H.atomic_radius_u
<Quantity(25.0, 'picometer')>

# with unit conversion
H.atomic_radius_u.to("bohr")
<Quantity(0.4724315311555698, 'bohr')>
```

Feedback is most welcome.
